code/modules/ruins/lavalandruin_code/techcult.dm

```diff
@@ -0,0 +1,592 @@
+	hoodtype = /obj/item/clothing/head/hooded/techpriest/armor
+
+/obj/item/clothing/head/hooded/techpriest/armor
+	name = "armored techpriest's hood"
+	desc = "An armored version of hood worn by followers of the machine god."
+	armor = list("melee" = 45, "bullet" = 10, "laser" = 25, "energy" = 25, "bomb" = 25, "bio" = 0, "rad" = 0, "fire" = 60, "acid" = 60)
+
+//Leader
+//Don't yell at me, it's super balanced, and in world of Lavaland everything is overpowered anyway.
+/obj/item/clothing/suit/hooded/techpriest/armor/lead
+	name = "blessed tech robes"
+	desc = "From the rage of the beast, Machine God protect us."
+	armor = list("melee" = 65, "bullet" = 40, "laser" = 40, "energy" = 40, "bomb" = 40, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 90)
+	hoodtype = /obj/item/clothing/head/hooded/techpriest/armor/lead
+
+/obj/item/clothing/suit/hooded/techpriest/armor/lead/Initialize()
+	. = ..()
+	AddComponent(/datum/component/anti_magic, FALSE, FALSE, TRUE, ITEM_SLOT_OCLOTHING)
+
+/obj/item/clothing/head/hooded/techpriest/armor/lead
+	name = "blessed tech hood"
+	desc = "From the weakness of the mind, Omnissiah set us free."
+	armor = list("melee" = 65, "bullet" = 40, "laser" = 40, "energy" = 40, "bomb" = 40, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 90)
+
+/obj/item/clothing/head/hooded/techpriest/armor/lead/Initialize()
+	. = ..()
+	AddComponent(/datum/component/anti_magic, FALSE, FALSE, TRUE, ITEM_SLOT_OCLOTHING)
+
+//Only the chosen ones can use it!
+/obj/item/clothing/suit/hooded/techpriest/armor/lead/equipped(mob/living/user, slot)
+	..()
+	if(!("Mechanicus" in user.faction))
+		to_chat(user, "<span class='warning'>The machine god will never allow this!</span>")
+		user.dropItemToGround(src, TRUE)
+		user.Paralyze(60)
+		user.Dizzy(300)
+	else
+		if(!(user.mind.holy_role))
+			to_chat(user, "<span class='warning'>Your time will come later.</span>")
+			user.dropItemToGround(src, TRUE)
+			user.Paralyze(40)
+			user.Dizzy(80)
+
+/***************** Spawners *****************/
+
+/obj/effect/mob_spawn/human/techcult
+	name = "Adept of the Machine Cult"
+	roundstart = FALSE
+	random = TRUE
+	death = FALSE
+	show_flavour = TRUE
+	mob_name = "tech priest"
+	icon = 'icons/obj/machines/sleeper.dmi'
+	icon_state = "sleeper"
+	short_desc = "You are a member of the machine cult on Lavaland."
+	flavour_text = "The flesh is weak and humans are fragile. You exist only to research universe and enchance abilities with power of science."
+	important_info = "Listen to your leader, help those in need and protect your religion."
+	outfit = /datum/outfit/techcult
+	assignedrole = "Tech Priest"
+
+/obj/effect/mob_spawn/human/techcult/special(mob/living/new_spawn)
+	new_spawn.grant_language(/datum/language/machine, TRUE, TRUE, LANGUAGE_MIND) //They have to implant robotic box instead of a tongue to speak this language though.
+
+/datum/outfit/techcult/post_equip(mob/living/carbon/human/H)
+	H.faction |= "Mechanicus"
```

Comment:
make sure that their armor won't reject them before they receive the mechanicus faction

Reply:
How dare you assume I didn't test it?

Yes, it works.